.github/workflows/test_on_push.yaml

```diff
@@ -27,8 +27,36 @@ jobs:
           python -m pip install pre-commit
           pre-commit run ruff
 
+  integration_tests:
+    needs: style
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.11"]
+        # Include MacOS M-series Runners
+        include:
+          - os: macos-14
+            python-version: "3.11"
+
```

Comment:
I'm mainly aiming to limit the `test_on_push` workflow time at this location. The unit tests run across all our supported Python versions, but I've selected a single Python version for the integration tests. This could be `3.12`, but I selected `3.11` as I suspect most users wouldn't immediately update to `3.12`. Open to other thoughts though.